### PR TITLE
[#Wave-Control: Add NetSuite]: Settings Configuration in NewDot: Import Custom Records/Lists

### PR DESCRIPTION
**Pull request by mananjadhav**
### Details

- Covers the implementation of [Custom segments/records and Custom lists](https://docs.google.com/document/d/1WubNv_VAv78IxG4FKsi9aS0pWESfWvUYbqBAAy5b4bc/edit#heading=h.pohqx06bjqcb).
- It only covers `Listing`, `View` , `Edit` and `Remove`. The add logic will be added in the future PR.

1. Collect workspace with accounting enabled.
2. `NetSuite` beta enabled for the user.  You can return true in the [canUseNetSuiteIntegration](https://github.com/Expensify/App/blob/6294e7d3462abe4124e6486de6702727b78a2c2c/src/libs/Permissions.ts#L51-L53) method.
3. Setup the netsuite connected workspace.
4. We might need to add a few records in `Custom segments/records` and `Custom lists` using OldDot as the screens for add are not ready yet.
5. While `custom lists` name field is a text input it's supposed to be a selection list. I'll add this along with the Add new PR.

### Fixed Issues

$ https://github.com/Expensify/App/issues/43437
PROPOSAL:

### Tests

- [x] Verify that no errors appear in the JS console

1. Open the app with the pre-setup workspace.
2. Go to `Accounting` from the LHS
3. Verify that you're able to see `Net Suite` in the active connections along with the `Oracle NetSuite` logo.
4. Click on the `Import` button.
5. Verify that the list of options appear as per the linked design doc.
6. Click on the `Custom segments/records` without any data.
7. You should see an the empty screen with the folder icon, and the text as following:

image

8. Now add a few records in `Custom segments/records` using OldDot.
9. Come back to the NewDot to the Import screen.
10. Click on the `Custom segments/records` with data now.
11. Verify that you can see the list of segments. Each segment should have the title `Custom segment/record` and the segmentName that you added.
12. Verify that you can see the description as the subtitle.
13. Click on any one of the records, you see see the view screen with all the fields: `Name`, `Internal ID`, `Script` and `Mapping`
14. Click on any one of the fields, and you should see the edit screen. For `mapping` you should see the dropdown with `Tags` and `Report` option. For the other fields it should be a text field.
15. The form should throw an error if the field is empty.
16. Update the value for these fields, and then click on `Save`.
17. It should update the field value and take you back to the record view screen.
18. Now to test delete, click on `Remove` and you should see a confirmation modal for remove.
19. Clicking on Remove should redirect back to the list screen and the record removed.
20. Repeat steps 6-19 for `Custom lists` in the import screen

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3069065/bb0ffe0f-644e-43d9-9300-3f325412dd94

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/3069065/4dbefebc-052d-4274-8b19-dd8edd895a60

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3069065/61c8d616-cff1-4f8c-aae9-0d3987b893b3

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/3069065/3e3e08ea-e384-4e7c-95af-c46db79927e1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

**List View with Empty**

https://github.com/Expensify/App/assets/3069065/94b835f8-3dc0-47ac-8cdd-ac6ed46348bf

**Update**

https://github.com/Expensify/App/assets/3069065/f631bf7f-80db-4d76-b422-541414effacf

**Remove**

https://github.com/Expensify/App/assets/3069065/7cea2453-e1ae-49d4-a335-cdce5f5ab90c

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/3069065/c34b5549-cec6-4996-a9da-f91ebe71eca7

</details>

### CODE REVIEW

File: src/languages/es.ts [hunk 1]
```diff
@@ -2442,8 +2442,40 @@ export default {
                 },
                 importTaxDescription: 'Importar grupos de impuestos desde NetSuite',
                 importCustomFields: {
-                    customSegments: 'Segmentos/registros personalizados',
-                    customLists: 'Listas personalizado',
+                    customSegments: {
+                        title: 'Segmentos/registros personalizado',
```
Comment (mananjadhav) [inline src/languages/es.ts:2446]: @yuwenmemon Can you verify the translations please?

Comment (yuwenmemon) [reply src/languages/es.ts:2446]: Plural form here
```suggestion
                        title: 'Segmentos/registros personalizados',
```

Custom Segments -> Segmentos personalizados 
Custom Records -> Registros personalizados
Custom Lists -> Listas personalizadas

That's the terminology from NetSuite.

Comment (yuwenmemon) [reply src/languages/es.ts:2446]: @zsgreenwald Can I enlist your help getting translations for these in #espanol-translations?

**Comment by mananjadhav** [comment]
@mollfpr Please ignore this PR.

@shubham1206agra @yuwenmemon For you review.

**Comment by mananjadhav** [comment]
Review for https://github.com/Expensify/App/pull/44942 already started. So closing this one out.

**Pull request closed without merging**